c7n/resources/account.py

```diff
@@ -1603,3 +1603,106 @@ class AccountCatalogEncryptionFilter(GlueCatalogEncryptionEnabled):
+class EMRBlockPublicAccessConfiguration(ValueFilter):
+    """Check for EMR block public access configuration on an account
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: get-emr-block-public-access-configuration
+                resource: account
+                filters:
+                  - type: emr-block-public-access-configuration
+    """
+
+    annotation_key = 'c7n:emr-block-public-access-configuration'
+    annotate = False  # no annotation from value filter
+    schema = type_schema('emr-block-public-access-configuration', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ("elasticmapreduce:GetBlockPublicAccessConfiguration",)
+
+    def process(self, resources, event=None):
+        self.augment([r for r in resources if self.annotation_key not in r])
+        return super().process(resources, event)
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client(
+            'emr', region_name=self.manager.config.region)
+
+        for r in resources:
+            try:
+                r[self.annotation_key] = client.get_block_public_access_configuration()
+                r[self.annotation_key].pop('ResponseMetadata')
+            except client.exceptions.NoSuchPublicAccessBlockConfiguration:
+                r[self.annotation_key] = {}
+
+    def __call__(self, r):
+        return super(EMRBlockPublicAccessConfiguration, self).__call__(r[self.annotation_key])
+
+
+@actions.register('set-emr-block-public-access-configuration')
+class PutAccountBlockPublicAccessConfiguration(BaseAction):
+    """Action to put/update the EMR block public access configuration for your
+       AWS account in the current region
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: set-emr-block-public-access-configuration
+                resource: account
+                filters:
+                  - type: emr-block-public-access-configuration
+                    key: BlockPublicAccessConfiguration.BlockPublicSecurityGroupRules
+                    value: False
+                actions:
+                  - type: set-emr-block-public-access-configuration
+                    BlockPublicAccessConfiguration:
+                        BlockPublicSecurityGroupRules: True
+                        PermittedPublicSecurityGroupRuleRanges:
+                            - MinRange: 22
+                              MaxRange: 22
+                            - MinRange: 23
+                              MaxRange: 23
+
+    """
+
+    schema = type_schema('set-emr-block-public-access-configuration',
+                         BlockPublicAccessConfiguration={"type": "object"},
+                         required=('BlockPublicAccessConfiguration',))
+    permissions = ("elasticmapreduce:PutBlockPublicAccessConfiguration",)
+    shape = 'PutBlockPublicAccessConfigurationInput'
```

Comment:
given how simple this schema is, we really shouldn't be using shape validate on it.

shape validation is a trade off, not a pancea.. it leads to poor docs and poor editor integration when using the jsonschema. it should really only be used for complex pass through schemas, else its much nicer to use jsonschema for end users.

Reply:
Makes sense, changed it to jsonshema now